src/app/machines/views/MachineList/MachineListTable/MachineCheckbox/MachineCheckbox.tsx

```diff
@@ -47,7 +102,18 @@ const MachineCheckbox = ({
       inputLabel={label}
       isChecked={isChecked ? Checked.Checked : Checked.Unchecked}
       isDisabled={allSelected || groupSelected}
-      onGenerateSelected={(checked) => {
+      onGenerateSelected={(checked, event) => {
+        window.getSelection()?.removeAllRanges();
```

Comment:
Comment why it's necessary might be useful.

Reply:
I'd suggest we move the base table checkbox range selection logic to `TableCheckbox` component for a better separation of concerns. This would include checking whether a shift key was pressed and prevention of the default text select behaviour.

Then you could just pass `(checked, isRange)` to the `MachineCheckbox` which has the logic specific to the Machine table.